adapta/storage/blob/s3_storage_client.py

```diff
@@ -132,15 +159,85 @@ def download_blobs(
         filter_predicate: Optional[Callable[[...], bool]] = None,
     ) -> None:
         """
-        Not yet implemented in S3 Client
+        Downloads blobs from S3 storage to a local path.
+
+        :param blob_path: Path to blob
+        :param local_path: Local path to download the blobs to
+        :param threads: Number of threads to use for the download
+        :param filter_predicate: Optional callable to filter blobs
+        :return:
+        """
+        s3_path = cast_path(blob_path)
+        blobs = self._s3_resource.Bucket(s3_path.bucket).objects.filter(Prefix=s3_path.path)
+        for blob in blobs:
+            if filter_predicate is None or filter_predicate(blob):
+                local_file_path = f"{local_path}/{blob.key}"
+                os.makedirs(os.path.dirname(local_file_path), exist_ok=True)
+                try:
+                    self._s3_resource.meta.client.download_file(s3_path.bucket, blob.key, f"{local_path}/{blob.key}")
+                except StorageClientError as error:
+                    raise RuntimeError(f"Error downloading blob: {error}") from error
+
+    def copy_blob(self, blob_path: DataPath, target_blob_path: DataPath, doze_period_ms: int = 0) -> None:
         """
-        raise NotImplementedError("Not yet implemented in S3StorageClient")
+        Copies a blob from one location to another in S3 storage.
 
-    def copy_blob(self, blob_path: DataPath, target_blob_path: DataPath, doze_period_ms: int) -> None:
+        :param blob_path: Path to the source blob
+        :param target_blob_path: Path to the target location
+        :param doze_period_ms: Not utilized for AWS operations, iIt is included to ensure interface compliance.
```

Comment:
```suggestion
        :param doze_period_ms: Not utilized for AWS operations, it is included to ensure interface compliance.
```

Reply:
Nice catch! Thank you for pointing it out 🙂